### PR TITLE
Support JSON credentials in log4net

### LINKED ISSUE #2314: GoogleStackdriverAppender doesn't allow for custom credentials
There is a `CredentialFile` property but please consider something like `CredentialJson` or a more generic `Credential`. That would be useful for loading the credential data for example from encrypted storage, DPAPI, or some vault rather than from a plain file on disk.

Thanks!

https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/master/apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender.cs#L204

### PR DESCRIPTION
**Pull request by chrisdunelm**
Fixes #2314

### CODE REVIEW

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.IntegrationTests/CredentialTests.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace Google.Cloud.Logging.Log4Net.IntegrationTests
+{
+    [SnippetOutputCollector]
+    [Collection(nameof(Log4NetFixture))]
+    public class CredentialTests
+    {
+        public CredentialTests(Log4NetFixture fixture) => _fixture = fixture;
+
+        private Log4NetFixture _fixture;
```
Comment (jskeet) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.IntegrationTests/CredentialTests.cs]: readonly

Comment (chrisdunelm) [reply apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net.IntegrationTests/CredentialTests.cs]: Oh yes, done.

File: apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs [hunk 2]
```diff
@@ -233,15 +229,28 @@ public int MaxMemoryCount
...
+        private string _credentialJson;
+        /// <summary>
+        /// JSON credential for authentication.
+        /// Not necessary if running on GCE or GAE or if the GOOGLE_APPLICATION_CREDENTIALS environment variable has been set.
+        /// Must not be set if <see cref="CredentialFile"/> is set.
+        /// </summary>
+        public string CredentialJson
+        {
+            get => _credentialJson;
+            set => _credentialJson = ThrowIfActivated(value, nameof(CredentialJson));
```
Comment (ILMTitan) [inline apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs]: Is there a reason `ThrowIfActivated` does not use `[CallerMemberName]`?

Comment (chrisdunelm) [reply apis/Google.Cloud.Logging.Log4Net/Google.Cloud.Logging.Log4Net/GoogleStackdriverAppender_Configuration.cs]: Prejudice against such magic?

**Pull request merged**